build/push-release-tarball.sh

```diff
@@ -5,12 +5,11 @@ set -euxo pipefail
 # This file uses `bash` and not `sh` due to the `time` builtin (the external
 # `time` is not available on CircleCI).
 
-BUCKET_NAME="binaries.cockroachdb.com"
```

Comment:
should be fine since `push-tagged-aws.sh` is the only caller and we have `-eu` on the script.
Another option would be to move all this to two functions inside `push-tagged-aws`, one to do all the tar/gz, the other to just copy.